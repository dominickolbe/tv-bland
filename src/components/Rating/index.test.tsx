import { shallow } from "enzyme";
import React from "react";
import { Rating } from "./";

describe("<Rating />", () => {
  it("renders snapshots, value 5", () => {
    const wrapper = shallow(<Rating value={5} withText={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders snapshots, value 7.9", () => {
    const wrapper = shallow(<Rating value={7.9} withText={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders snapshots, value undefined", () => {
    const wrapper = shallow(<Rating value={undefined} withText={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders snapshots, withText false", () => {
    const wrapper = shallow(<Rating value={3.33} withText={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
