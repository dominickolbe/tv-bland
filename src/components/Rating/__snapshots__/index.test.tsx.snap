// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating /> renders snapshots, value 5 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Rating))
    precision={0.5}
    readOnly={true}
    size="small"
    value={2.5}
  />
  <WithStyles(ForwardRef(Typography))
    component="div"
    variant="body1"
  >
    <Styled(MuiBox)
      fontWeight="fontWeightBold"
      ml={2}
    >
      2.5
       / 5
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Typography))>
</div>
`;

exports[`<Rating /> renders snapshots, value 7.9 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Rating))
    precision={0.5}
    readOnly={true}
    size="small"
    value={3.95}
  />
  <WithStyles(ForwardRef(Typography))
    component="div"
    variant="body1"
  >
    <Styled(MuiBox)
      fontWeight="fontWeightBold"
      ml={2}
    >
      4.0
       / 5
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Typography))>
</div>
`;

exports[`<Rating /> renders snapshots, value undefined 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Rating))
    precision={0.5}
    readOnly={true}
    size="small"
    value={null}
  />
  <WithStyles(ForwardRef(Typography))
    component="div"
    variant="body1"
  >
    <Styled(MuiBox)
      fontWeight="fontWeightBold"
      ml={2}
    >
      -
       / 5
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Typography))>
</div>
`;

exports[`<Rating /> renders snapshots, withText false 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Rating))
    precision={0.5}
    readOnly={true}
    size="small"
    value={1.665}
  />
</div>
`;
